src/ui/src/wds/WdsIcons/LucideIcon.vue

```diff
@@ -0,0 +1,48 @@
+<!-- eslint-disable vue/no-v-html -->
+
+<template>
+	<span :style="{ display: 'contents' }" v-html="rawHtml" />
+</template>
+
+<script setup lang="ts">
+import { nextTick, onMounted, ref, useAttrs, watch } from "vue";
+import { createIcons, icons } from "lucide";
+
+defineOptions({
+	inheritAttrs: false,
+});
+
+const props = defineProps({
+	name: { type: String, required: true },
+});
+
+const attrs = useAttrs();
+
+const rawHtml = ref("");
+
+function renderIcon(name: string) {
+	rawHtml.value = `<i data-lucide="${name}"></i>`;
+
+	nextTick(() => {
+		createIcons({
+			icons,
+			attrs: {
+				width: "1em",
+				height: "1em",
```

Comment:
I think it's better to have `width` & `height` as props. Also, we don't use `em`

```suggestion
const props = defineProps({
	name: { type: String, required: true },
	size: { type: String, required: false, default: '12px' },
});

const rawHtml = ref("");

function renderIcon(name: string) {
	rawHtml.value = `<i data-lucide="${name}"></i>`;

	nextTick(() => {
		createIcons({
			icons,
			attrs: {
				width: props.size,
				height: props.size,
```

Reply:
Yeah, that one is tricky. I used em intentionally to match the behavior of Material icons, since they are essentially icon fonts that scale with the font-size of the parent. You can check that `material-symbols-outlined` class uses font-size: 100%, so I wanted to preserve that dynamic scaling for SVG icons too.

But I'm totally fine switching to a fixed size prop (like px) if that's the direction we want to take. But we would need to make sure it doesn't break any cases where icon size was previously inherited via font-size.

Reply:
Good point, so I guess we can keep your implementation :+1: